backend/human.js

```diff
@@ -110,35 +121,105 @@ module.exports = class extends Player {
   }
-  pick(index) {
-    const pack = this.packs.shift();
-    const card = pack.splice(index, 1)[0];
+  static pick(index) {
 
-    this.draftLog.pack.push( [`--> ${card.name}`].concat(pack.map(x => `    ${x.name}`)) );
-    this.updateDraftStats(this.draftLog.pack[ this.draftLog.pack.length-1 ], this.pool);
+    const log_pick = (card, pack) => {
+      this.draftLog.pack.push(
+        [`--> ${card.name}`].concat(
+          pack.map(x => `    ${x.name}`)
+        )
+      );
+      this.updateDraftStats(
+        this.draftLog.pack[ this.draftLog.pack.length-1 ],
+        this.pool
+      );
+    };
 
-    let pickcard = card.name;
-    if (card.foil === true)
-      pickcard = "*" + pickcard + "*";
+    const pick_a_card_do = (card) => {
+      let foil_name_maybe = card.name;
+      if (card.foil === true)
+        foil_name_maybe = "*" + foil_name_maybe + "*";
+      this.pool.push(card);
+      this.picks.push(foil_name_maybe);
+      this.send("add", card);
+    };
 
-    this.pool.push(card);
-    this.picks.push(pickcard);
-    this.send("add", card);
+    const pick_a_card = (index) => {
+      const pack = this.packs[0];
+      const card = pack.splice(index, 1)[0];
+      log_pick(card, pack);
+      pick_a_card_do(card);
+    };
 
-    let [next] = this.packs;
-    if (!next)
-      this.time = 0;
-    else
-      this.sendPack(next);
+    const burn_a_card = (index) => {
+      const pack = this.packs[0];
+      const card = pack.splice(index, 1)[0];
+      log_pick(card, pack);
+    };
+
+    const keep_the_pack = () => {
+      const pack = this.packs[0];
+      if (!pack)
+        this.time = 0;
+      else
+        this.sendPack(pack);
+      this.autopick_indexes.splice(0, 1);
+      this.emit("keep", pack);
+      return Symbol("ok");
+    };
+
+    const send_next_pack_to_frontend_maybe = () => {
+      let [next] = this.packs;
+      if (!next)
+        this.time = 0;
+      else
+        this.sendPack(next);
+      return Symbol("ok");
+    };
+
+    const pass_the_pack = () => {
+      const pack = this.packs.shift();
+      this.autopick_indexes = this.callbacks.default_pick_indexes();
+      send_next_pack_to_frontend_maybe();
+      this.emit("pass", pack);
+      return Symbol("ok");
+    };
+
+    const end_round = pass_the_pack;
+
+    const pick_do = () => {
+      const pack = this.packs[0];
+      const indexes = this.autopick_indexes;
+      const picks_total = this.callbacks.default_pick_indexes().length;
+      if (indexes.length === picks_total) {
+        pick_a_card(index);
+        if (pack.length === 0)
+          return end_round();
+        else if (picks_total === 1)
+          return pass_the_pack();
+        else
+          return keep_the_pack();
+      } else if (indexes.length === 1) {
+        burn_a_card(index);
+        return pass_the_pack();
+      } else {
+        burn_a_card(index);
+        if (pack.length === 0)
+          return pass_the_pack();
```

Comment:
Gah, this should say `end_the_round()` for readability, but they are aliases anyway. Sorry for inconsistency, but I'm too lazy to figure out how to amend pull requests without creating new commits. I hope that's not a big deal.

Reply:
A new commit is just fine!
GitHub allows to squash them back to one on merge.